docs/CppInteroperability/Roadmap.md

```diff
@@ -0,0 +1,280 @@
+
+Note: in the example above, “start” is a pointer type. Using this pointer returned by the “begin” method is unsafe, but the type of start itself is not unsafe. In other words, safety restrictions need not be applied to pointer types themselves but rather their unsafe uses.
+
+Editorial note: the following three paragraphs have a lot of overlap with the above section and should be folded together somehow. 
+
+C++ often projects the storage of owned types. C++ is able to tie the lifetime of the projection to the source using lexcal scopes. Because there is a well-defined, lexical point in which objects are destroyed, C++ users can reason about projection’s lifetimes. While these safety properties are less formal than Swift, they are safety properties none-the-less, and they are a model that works in C++. 
+
+This model cannot be adopted in Swift, however, because the the same lexical lifetime model does not exist. Further, projections of owned types are completely foreign concept in Swift, meaning users aren’t used to programming in terms of this model, and may not be aware of the added (implicit) constraints (that is, when objects are destroyed). Swift’s language model is such that returning projections from a copied value, even in smaller lexical scope, should be safe. In order to allow projections of owned types, this assumption must be broken, or C++ interoperability must take advantage of Swift ownership features to associate the lifetime of the projection to the source. 
+
+The following example highlights the case described above:
+
+```
+func getCString(str: std.string) -> UnsafePointer<CChar> { str.c_str() }
+```
+
+The above function returns a dangling reference to `str`‘s inner storage. In C++, it is assumed that the programmer understands this is a bug, and generally would be expected to take `str` by reference. This is not the case in Swift. To represent this idiomatically in Swift, the lifetimes must be associated through a projection. Using the tools provided in the ownership manifesto this would mean yielding the value returned by `c_str` out of a [generalized accessor](https://github.com/apple/swift/blob/main/docs/OwnershipManifesto.md#generalized-accessors)(resulting in an error when the pointer is returned). 
+ 
```

Comment:
It think it is worth calling out that we would be providing annotations to import projections and also, perhaps, distinguish the following cases: 
- _The method looks like a projection but is not really one_ e.g. it may be returning a reference/pointer it was created with like `StringRef.getValue()` etc. In this case, it is safe to use it. So the annotation can read like `@has_indepedent_lifetime` to capture this fact and allow such functions. 
- _ The method is really a projection_: in this case we provide an `unsafe_import` attribute that renames the function and also comes with the expectation that the user is responsible for managing its lifetime. This is because, it  may be necessary for programmers to write ergonomic overlays for C++ types. We expect that our users use `@_noImplicitCopy` annotation  and `withExtendingLifetime` and other language affodances to carefully use these APIs in those cases.